### ISSUE TITLE
compile fails with error: cannot convert from 'hx::ObjectPtr<...>' to 'int'

### ISSUE THREAD

**Issue by nulld** [HaxeFoundation/hxcpp#583]
if rewrite this code without `using` everething works

minimal reproducing code:
Test.hx:
```
using StaticExt;

class Context
{
    public var storage:Map<String, Dynamic>;
    public function new()
    {
        storage = new Map<String, Dynamic>();
    }
}

class ActionsContainer
{
    var actions:Array<Void->Void>;

    public function new(actions:Array<Void->Void>)
    {
        this.actions = actions;
    }
}

class Test {

    public static function main() {
        var context = new Context();
        configure(context);
    }

    static function configure(context:Context):ActionsContainer
    {
        var fn:Void->Void = context.getId.bind("some");
        return new ActionsContainer([fn]);
    }
}
```

StaticExt.hx:
```
import Test.Context;

class StaticExt
{
    public static function getId(ctx:Context, name:String):Int
    {
        return ctx.storage.get(name);
    }
}
```
build.hxml
```
-cp .
-cpp build
-main Test
```

**Comment by nadako** [comment]
Could be related to #6121 (because of `bind` usage), but I can't reproduce it on latest haxe/hxcpp from git.

**Comment by hughsando** [comment]
Yes, I think this should be fixed on the 3.4.2.bugfix branch.

**Comment by nulld** [comment]
Thanks, so we can close this. For now, I returned to haxe 3.4.1. and it works on it.
